clang/lib/DPCT/CMakeLists.txt

```diff
@@ -77,6 +77,20 @@ add_custom_target(dpct_helper_headers_and_inc
   DEPENDS ${PROCESS_FILES_OUTPUT}
   )
 
+# Handle API mapping cases header
+set(API_MAPPING_REGISTER_FOLDER ${CMAKE_BINARY_DIR}/tools/clang/include/clang/DPCT/)
+file(GLOB SRC_FILES ../../examples/DPCT/*/*.cu)
+file(WRITE ${API_MAPPING_REGISTER_FOLDER}APIMappingRegister.def "")
+foreach(FILE ${SRC_FILES})
+  file(READ ${FILE} FILE_STR)
+  string(REGEX REPLACE ".*/(.*).cu$" "\\1" API_NAME ${FILE})
+  file(APPEND ${API_MAPPING_REGISTER_FOLDER}APIMappingRegister.def
+  "registerEntry(\"" "${API_NAME}" "\",\nR\"(" "${FILE_STR}" ")\");\n")
```

Comment:
how about add logic to check if ${API_NAME} is unique, if not, report error during build time.

Reply:
done